plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -135,8 +135,11 @@ public TableFunctionProcessorState process()
 
     private TableFunctionProcessorState processCdfFile()
     {
-        Page page = deltaLakePageSource.getNextPage();
-        if (page != null) {
+        while (!deltaLakePageSource.isFinished()) {
+            Page page = deltaLakePageSource.getNextPage();
+            if (page == null) {
+                continue;
```

Comment:
The idea behind `CPS.getNextPage` returning nullable is to break procesisng producing a page takes too long.

I think we should yield here. There are 3 ways we can go about  it

- returning `TableFunctionProcessorState.Blocked.blocked(NOT_BLOCKED)`
- returning `TableFunctionProcessorState.Processed.produced(EMPTY_PAGE)`
- introducing new `TableFunctionProcessorState.Yielded`

let's go with `TableFunctionProcessorState.Processed.produced(EMPTY_PAGE)` for a quick fix
and let's follow up with a PR introducing `TableFunctionProcessorState.Yielded` state